process/block/shardblock.go

```diff
@@ -742,6 +745,38 @@ func (sp *shardProcessor) CommitBlock(
 		headerMeta.GetNonce(),
 	)
 
+	headerInfo := bootstrapStorage.BootstrapHeaderInfo{
+		ShardId: header.ShardId,
+		Nonce:   header.Nonce,
+		Hash:    headerHash,
+	}
+
+	log.Debug("validator info on block ",
+		"nonce", header.Nonce,
+		"validator root hash", core.ToB64(header.ValidatorStatsRootHash))
+
+	sp.mutProcessedMiniBlocks.RLock()
+	log.Debug("processed mini blocks on commit block")
+	for key, value := range sp.processedMiniBlocks {
+		log.Debug("processed",
+			"meta block hash", []byte(key))
+
+		for miniBlockHash := range value {
+			log.Debug("processed",
+				"mini block hash", []byte(miniBlockHash))
+
+		}
+	}
+
+	processedMiniBlockBytes, errNotCritical := json.Marshal(sp.processedMiniBlocks)
+	sp.mutProcessedMiniBlocks.RUnlock()
+
+	if errNotCritical != nil {
+		log.Debug("commit block",
+			"cannot marshal processed mini block map", errNotCritical.Error())
+	}
+	sp.prepareDataForBootStorer(headerInfo, header.Round, finalHeaders, finalHeadersHashes, processedMiniBlockBytes)
```

Comment:
`finalHeaders, finalHeadersHashes` can be grouped in a separate type. We would have less parameters and we would eliminate the risk of having a mismatch between them